src/Extensions/Hosting.Services/Internal/ServiceContextAccessor.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Fabric;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services
+{
+	internal class ServiceContextAccessor<TContext> : IServiceContextAccessor<TContext>
+		where TContext : ServiceContext
+	{
+		public ServiceContextAccessor(TContext? context = null)
+		{
+			m_serviceContext = context;
+			m_actions = new LinkedList<Action<TContext>>();
```

Comment:
Is `LinkedList` definitely the right object here? Are we expecting additions that would be beyond the scope of what we can expect `List` to handle?

Reply:
My reasoning behind it was that we won't need by element access and will require only single iteration through all of the objects plus multiple adding operations, so `LinkedList` should be more efficient here.

Reply:
In that case `LinkedList` does sound more efficient. I was surprised because almost no one uses it. If you do know the final size of the list, you can specify a default capacity in which case the perf of `List` is likely marginally better and it was consume less space.

Reply:
Feel free to resolve this conversation after reading my comment above.

Reply:
Thank you for the response, I don't know end list size that's another reason for using LinkedList